tools/clang/lib/Sema/SemaHLSLDiagnoseTU.cpp

```diff
@@ -520,6 +520,18 @@ void hlsl::DiagnoseTranslationUnit(clang::Sema *self) {
               << hullPatchCount.value();
         }
       }
+      for (const auto *param : pPatchFnDecl->params())
+        if (ContainsLongVector(param->getType()))
+          self->Diag(param->getLocation(),
+                     diag::err_hlsl_unsupported_long_vector)
+              << DXIL::kDefaultMaxVectorLength
+              << "patch constant function parameters";
+
+      if (ContainsLongVector(pPatchFnDecl->getReturnType()))
+        self->Diag(pPatchFnDecl->getLocation(),
+                   diag::err_hlsl_unsupported_long_vector)
+            << DXIL::kDefaultMaxVectorLength
+            << "patch constant function return type";
```

Comment:
curious, why can't long vectors be used in patch functions? I tried to find a reference in the spec and didn't see anything. Am I missing something? Just trying to connect some dots to help learning.